Marlin/Marlin_main.cpp

```diff
@@ -6507,6 +6520,9 @@ inline void gcode_T(uint8_t tmp_extruder) {
   #endif
 
   #if HOTENDS > 1
+  
+  // Are there variates of T who do not need to be homed before?
+  if (axis_unhomed_error(true, true, true)) return;
```

Comment:
If you have a tool with no offsets, then you should be able to switch tools before homing without issue. I don't see any reason to disable tool-change, in particular, but I see the point if the nozzles might move. I think the tool-change should still happen, but it should throw the unhomed error, and set the `no_move` flag so the tool doesn't try to crash into anything.

In a corporate design lab we'd ultimately make the machine home on startup as a standard behavior. And if it didn't succeed we'd go to the kill screen. But due to the nature of 3D printers and how people use them, I guess we have to allow for these machines to be "unknown" at any time.